resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/v1/Client.scala

```diff
@@ -50,7 +50,8 @@ private[spark] class Client(
   private val launchTime = System.currentTimeMillis
   private val appName = sparkConf.getOption("spark.app.name")
     .getOrElse("spark")
-  private val kubernetesAppId = s"$appName-$launchTime".toLowerCase.replaceAll("\\.", "-")
+  private val kubernetesAppId = sparkConf.getOption("spark.kubernetes.driver.pod.name")
```

Comment:
the pod name `spark.kubernetes.driver.pod.name` and the app ID `spark.app.id` are slightly different -- I think the change on this line should be pulling from `spark.app.id` if we want to make the app ID configurable.

Given that this PR is about the pod name and not the app ID, maybe drop the change on this line and leave just the pod name configurable?